components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,178 @@
+        }
+        // we only want this behavior when showDropDown changes to true.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [showDropDown]);
+
+    const onKeyDown = useMemo(
+        () => (e: React.KeyboardEvent) => {
+            if (!showDropDown) {
+                return;
+            }
+            if (e.key === "ArrowDown") {
+                e.preventDefault();
+                let idx = filteredOptions.findIndex((e) => e.id === selectedElementTemp);
+                while (idx++ < filteredOptions.length - 1) {
+                    const candidate = filteredOptions[idx];
+                    if (candidate.isSelectable) {
+                        setSelectedElementTemp(candidate.id);
+                        document.getElementById(candidate.id)?.scrollIntoView({ behavior: "smooth", block: "nearest" });
+                        return;
+                    }
+                }
+                return;
+            }
+            if (e.key === "ArrowUp") {
+                e.preventDefault();
+                let idx = filteredOptions.findIndex((e) => e.id === selectedElementTemp);
+                while (idx-- > 0) {
+                    const candidate = filteredOptions[idx];
+                    if (candidate.isSelectable) {
+                        setSelectedElementTemp(candidate.id);
+                        document.getElementById(candidate.id)?.scrollIntoView({ behavior: "smooth", block: "nearest" });
+                        return;
+                    }
+                }
+                return;
+            }
+            if (e.key === "Escape") {
+                setShowDropDown(false);
+                e.preventDefault();
+            }
+            if (e.key === "Enter" && selectedElementTemp && filteredOptions.some((e) => e.id === selectedElementTemp)) {
+                e.preventDefault();
+                props.onSelectionChange(selectedElementTemp);
+                setShowDropDown(false);
+            }
+        },
+        [filteredOptions, props, selectedElementTemp, showDropDown],
+    );
+
+    const handleBlur = useMemo(
+        () => () => {
+            // postpone a little, so it doesn't fire before a click event for the main element.
+            setTimeout(() => setShowDropDown(false), 100);
+        },
+        [setShowDropDown],
+    );
+
+    const toggleDropDown = useMemo(
+        () => () => {
+            setShowDropDown(!showDropDown);
+        },
+        [setShowDropDown, showDropDown],
+    );
+
+    return (
+        <div onKeyDown={onKeyDown} className="relative flex flex-col">
+            <div
+                className={
+                    "h-16 bg-gray-100 dark:bg-gray-800 flex items-center px-2 " +
+                    (showDropDown
+                        ? "rounded-t-lg"
+                        : "rounded-lg hover:bg-gray-200 dark:hover:bg-gray-700 cursor-pointer")
```

Comment:
When using the dropdown, it felt weird that although after opening it, I can still interact with its header, there is no more a pointer cursor. 

```suggestion
                    "h-16 bg-gray-100 dark:bg-gray-800 flex items-center px-2 cursor-pointer" +
                    (showDropDown
                        ? "rounded-t-lg"
                        : "rounded-lg hover:bg-gray-200 dark:hover:bg-gray-700")
```